edx_when/api.py

```diff
@@ -296,14 +364,17 @@ def get_overrides_for_user(course_id, user):
         yield {'location': udate.content_date.location, 'actual_date': udate.actual_date}
 
 
-def set_date_for_block(course_id, block_id, field, date_or_timedelta, user=None, reason='', actor=None):
+def set_date_for_block(
+        course_id, block_id, field, date_or_timedelta,
+        user=None, reason='', actor=None, published_version=None):
```

Comment:
FYI: while the `published_version` is saved on each `ContentDate` object, it's not used as a filter when querying course dates (though it *is* returned for each block).